book/src/dev/rfcs/0005-state-updates.md

```diff
@@ -300,11 +304,13 @@ Remove the lowest height block of the non-finalized portion of a chain.
 
 2. Update cumulative data members
     - Remove the block's hash from `self.height_by_hash`
-    - for each `transaction` in `block`
-      - remove `transaction.hash` from `tx_by_hash`
-    - Remove new utxos from `self.utxos`
-    - Remove the anchors from the appropriate `self.<version>_anchors`
-    - Remove the nullifiers from the appropriate `self.<version>_nullifiers`
+    - Subtract work from `self.partial_cumulative_work`
+    - For each `transaction` in `block`
+      - Remove `transaction.hash` from `tx_by_hash`
+      - Remove created utxos from `self.created_utxos`
+      - Remove spent utxos from `self.spent_utxos`
+      - Remove the anchors from the appropriate `self.<version>_anchors`
```

Comment:
As noted above this is not really pre-transaction, there should be one each for Sprout and Sapling per block